src/server/server.ts

```diff
@@ -150,7 +150,12 @@ export const createApp = <E extends Env>(options: BaseServerOptions<E>): Hono<E>
       })
 
       const middlewareFile = Object.keys(MIDDLEWARE_FILE).find((x) => {
-        const replacedDir = dir.replaceAll('[', '\\[').replace(']', '\\]')
```

Comment:
@yusukebe - I'm quite sure that `routes/content/[category]/[id]/_middleware.ts` would not have worked.

Reply:
Ah, you are right!